MedtrumKit/Encryption/ReadPacket.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  ReadPacket.swift
+//  MedtrumKit
+//
+//  Created by Bastiaan Verhaar on 27/02/2025.
+//
+
+class ReadPacket {
```

Comment:
Something to consider, In AAPS we had this indeed, In insubox I merged the serialisation of the incomming messages in the basePacket as well.

See:
https://github.com/jbr7rr/insuBox/blob/8f217de4df784e86a49ee380fc07968353325e9b/lib/pump/medtrum_bt/comm/packets/MedtrumBasePacket.cpp#L29


It can make the handling of incoming messages, more straightforward, see:
https://github.com/jbr7rr/insuBox/blob/8f217de4df784e86a49ee380fc07968353325e9b/lib/pump/medtrum_bt/MedtrumBTDevice.cpp#L92

Reply:
Done!